core-api/src/main/java/com/optimizely/ab/internal/DefaultLRUCache.java

```diff
@@ -0,0 +1,147 @@
+/**
+ *
+ *    Copyright 2022, Optimizely
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.internal;
+
+import com.optimizely.ab.annotations.VisibleForTesting;
+
+import java.util.*;
+
+public class DefaultLRUCache<T> implements LRUCache<T> {
+
+    private final Object lock = new Object();
+
+    private Integer maxSize;
+
+    private Long timeoutMillis;
+
+    final LinkedList<ItemWrapper> linkedList = new LinkedList<>();
+
+    final Map<String, ItemWrapper> hashMap = new HashMap<>();
+
+    public DefaultLRUCache() {
+        this.maxSize = DEFAULT_MAX_SIZE;
+        this.timeoutMillis = (long) (DEFAULT_TIMEOUT_SECONDS * 1000);
+    }
+
+    public void setMaxSize(Integer size) {
+        Integer sizeToSet = size;
+        if (size < 0 ) {
+            sizeToSet = DEFAULT_MAX_SIZE;
```

Comment:
either way is fine, but let's use "sizeToSet = 0" for all SDKs.

Reply:
Done! Defaulting to zero now when max size is negative